src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/RedirectedBasicWriter.cs

```diff
@@ -70,7 +72,39 @@ public override void WriteHtmlAttribute(CSharpRenderingContext context, HtmlAttr
 
         public override void WriteHtmlContent(CSharpRenderingContext context, HtmlContentIRNode node)
         {
-            _previous.WriteHtmlContent(context, node);
+            if (context.Options.DesignTimeMode)
```

Comment:
We don't. But that is not up to this writer to decide. We don't know here what type of writer `_previous` is.